lib/upstream/core.js

```diff
@@ -91,6 +91,19 @@ class Core {
     });
   }
 
+  destroy(fourfour, onComplete) {
+    return this._url((err, url) => {
+      if (err) return onComplete(err);
+      request.del({
+          url: `${url}/views/${fourfour}`,
+          headers: this._headers(),
+          json: true
+        })
+        .on('response', this._onResponseStart(onComplete))
```

Comment:
Why do we need both handlers here? We end up checking the response code in `_onResponseStart`. What are the different scenarios here?